process_report/process_report.py

```diff
@@ -58,7 +58,14 @@ def load_old_pis(old_pi_file):
     return old_pi_dict
 
 
+def dump_old_pis(old_pi_file, old_pi_dict: dict):
+    with open(old_pi_file, "w") as f:
+        for pi, first_month in old_pi_dict.items():
+            f.write(f"{pi},{first_month}\n")
+
+
 def is_old_pi(old_pi_dict, pi, invoice_month):
+    # If we're processing old invoices, this will flag PIs who were new *back then* as old. But this would only be the case if future PIs appeared in past invoices???
```

Comment:
- Do not write lines this long, break the comment into multiple line comments.
- Comments are not a place to have a dialogue with yourself. An issue with the code should be flagged as an issue, or resolved.
- You can easily turn this check from a `!=` to comparing if the date is in the past or future. 
  - Write a unit test for it.